packages/video_player/video_player/lib/video_player.dart

```diff
@@ -212,12 +219,16 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
         dataSourceType = DataSourceType.file,
         package = null,
         formatHint = null,
+        httpHeaders = null,
         super(VideoPlayerValue(duration: Duration.zero));
 
   /// The URI to the video file. This will be in different formats depending on
   /// the [DataSourceType] of the original video.
   final String dataSource;
 
+  /// HTTP headers used for the request to the [dataSource].
+  final Map<String, String>? httpHeaders;
```

Comment:
We generally prefer to make collections non-nullable and defaulting to empty, so that we don't have both null and empty as codepaths that mean the same thing (except in cases where null and empty actually mean different things, but that's not true here). It avoids bugs where, e.g., one platform implementation handles them the same, but one fails with one or the other (or treats them differently on accident).

Reply:
As pigeon doesn't currently support non-nullable fields, generated fields will be still nullable. If we make headers non-nullable in VideoPlayerController class, should we check for nulls in platform code then?

Reply:
Well I made it non-nullable and empty by default. Also I kept check for null and added check for empty in platform code. So the default behavior is untouched (don't call function setting headers in native code if no headers were passed). And also, if null will git into native side somehow, it will be also ignored